File: backend/routes/protected.py
from fastapi import APIRouter, Depends

from middleware.auth import (
    authenticate_token,
    authorize_role
)


router = APIRouter(
    prefix="/api",
    tags=["Protected Routes"]
)


@router.get("/profile")
def profile(
    current_user=Depends(authenticate_token)
):
    return {
        "success": True,
        "message": "Protected profile accessed",
        "user": current_user
    }


@router.get("/dashboard")
def dashboard(
    current_user=Depends(authenticate_token)
):
    return {
        "success": True,
        "message": "Dashboard accessed successfully",
        "user": current_user
    }


@router.get("/admin")
def admin_dashboard(
    current_user=Depends(
        authorize_role("admin")
    )
):
    return {
        "success": True,
        "message": "Admin dashboard accessed",
        "user": current_user
    }

File: routes/protected.js
const express = require("express");

const {
    authenticateToken,
    authorizeRole
} = require("../middleware/authMiddleware");

const router = express.Router();


// ======================================================
// Protected user route
// ======================================================

router.get("/profile", authenticateToken, (req, res) => {

    return res.status(200).json({
        success: true,
        message: "Protected profile accessed successfully",
        user: req.user
    });

});


// ======================================================
// Protected dashboard
// Any authenticated user can access
// ======================================================

router.get("/dashboard", authenticateToken, (req, res) => {

    return res.status(200).json({
        success: true,
        message: "Welcome to the protected dashboard",
        user: req.user
    });

});


// ======================================================
// Admin-only route
// ======================================================

router.get(
    "/admin",
    authenticateToken,
    authorizeRole("admin"),
    (req, res) => {

        return res.status(200).json({
            success: true,
            message: "Admin resource accessed successfully",
            user: req.user
        });

    }
);


module.exports = router;